### PR TITLE
convert all tours

### PR DESCRIPTION
**Pull request by Gorash**

### CODE REVIEW

File: addons/website_sale/static/src/js/website_sale_tour_buy.js [hunk 1]
```diff
@@ -29,9 +29,12 @@ tour.register('shop_buy_product', { test: true, url: '/shop' },
             trigger: '#product_detail form[action^="/shop/cart/update"] .btn',
         },
         {
+            content: "wait to add suggested because website_sale_options change tour",
```
Comment (Yenthe666) [inline addons/website_sale/static/src/js/website_sale_tour_buy.js]: @Gorash this is a weird sentence? It confuses me.

Comment (Gorash) [inline addons/website_sale/static/src/js/website_sale_tour_buy.js]: A bug in the new tour, I look to fix because without fix we need to add stupid step

File: addons/point_of_sale/static/src/js/tests.js [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-            title: 'buy ' + product_name,
-            element: '.product-list .product-name:contains("' + product_name + '")',
+            content: 'buy ' + product_name,
+            trigger: '.product-list .product-name:contains("' + product_name + '")',
         }, {
-            title: 'the ' + product_name + ' have been added to the order',
-            waitFor: '.order .product-name:contains("' + product_name + '")',
+            content: 'the ' + product_name + ' have been added to the order',
+            trigger: '.order .product-name:contains("' + product_name + '")',
+            run: function () {}, // it's a check
```
Comment (qsm-odoo) [inline addons/point_of_sale/static/src/js/tests.js]: How can all of these lines work ? The step trigger is found (for check), then the run method is called, which does nothing... so nothing happen and the tour should be stuck ?

File: addons/web/static/test/x2many.js [hunk 3]
```diff
@@ -1,610 +0,0 @@
-odoo.define('web.test.x2many', function (require) {
```
Comment (qsm-odoo) [inline addons/web/static/test/x2many.js]: The x2many file move should be in a separated commit, without any modification

File: addons/web_editor/static/src/js/tour.rte.js [hunk 4]
```diff
@@ -1,408 +1,337 @@
...
+                content: "simulate triple click and change text bg-color",
+                trigger: '#editable_area > section .row > div:first',
+                run: function () {
+                    var $h1 = $('h1', this.$anchor);
+                    $.summernote.core.range.create($h1[0].firstChild, 0, $('p', this.$anchor)[0], 0).select();
+                    click_event($h1[0], 'mouseup');
+                }
+            }, {
+                content: "change text bg-color after triple click",
+                trigger: '.note-popover button[data-event="color"]:visible',
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/tour.rte.js]: :visible at the end of a trigger/extra_trigger are useless

File: addons/web_tour/models/ir_http.py [hunk 5]
```diff
@@ -6,6 +6,5 @@ class Http(models.AbstractModel):
 
     def session_info(self):
         result = super(Http, self).session_info()
-        if result['is_superuser']:
-            result['web_tours'] = request.env['web_tour.tour'].get_consumed_tours()
```
Comment (qsm-odoo) [inline addons/web_tour/models/ir_http.py]: I am not sure about this, but it should at least be in a separated commit (with everything else which concerns the consumed tours loading)

File: addons/web_tour/static/src/js/tour_manager.js [hunk 6]
```diff
@@ -34,6 +34,9 @@ var RunningTourActionHelper = core.Class.extend({
     drag_and_drop: function (to, element) {
         this._drag_and_drop(this._get_action_values(element), to);
     },
+    keydown: function (keyCodes, element) {
+        this._keydown(this._get_action_values(element), keyCodes.split(/\s*[, ]\s*/));
```
Comment (qsm-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: /[,\s]+/ does the same thing I think

File: addons/web_tour/static/src/js/tour_manager.js [hunk 7]
```diff
@@ -52,21 +55,27 @@ var RunningTourActionHelper = core.Class.extend({
     },
     _click: function (values) {
         var href = values.$element.attr("href");
-        if (href && href.length && href[0] !== "#" && values.$element.is("a")) {
+        if (href && href.length && href[0] !== "#" && values.$element.is("a") && href !== 'javascript:void(0)') {
             window.location.href = href;
         } else {
-            values.$element.mousedown().mouseup().click();
+            values.$element.trigger("mouseenter").mousedown().mouseup().click().trigger("mouseleave");
```
Comment (qsm-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: Maybe use mouseenter() mouseleave() to be consistent with the 3 other calls ?

File: addons/web_tour/static/src/js/tour_manager.js [hunk 8]
```diff
@@ -52,21 +55,27 @@ var RunningTourActionHelper = core.Class.extend({
...
         }
     },
     _text: function (values, text) {
         this._click(values);
 
         text = text || "Test";
-        if (values.consume_event === "input") {
-            values.$element.val(text).trigger("input");
+        if (values.$element.is('select')) {
```
Comment (qsm-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: I would put this if section inside the else selection. So if (... === "input) {...} else { if (... is select) {} else {} }

File: addons/web_tour/static/src/js/tour_manager.js [hunk 9]
```diff
@@ -52,21 +55,27 @@ var RunningTourActionHelper = core.Class.extend({
...
+            values.$element.find('option').removeProp('selected')
+                .filter(function () { return $(this).val() == text; })
+                .prop('selected', 'selected');
+            values.$element.mousedown().mouseup().click();
+        } else if (values.consume_event === "input") {
+            values.$element.trigger('keydown').val(text).trigger('keyup').trigger("input");
         } else {
             values.$element.text(text);
         }
+        values.$element.trigger("change");
```
Comment (qsm-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: trigger change on all dom elements ?

File: addons/web_tour/static/src/js/tour_manager.js [hunk 10]
```diff
@@ -34,6 +34,9 @@ var RunningTourActionHelper = core.Class.extend({
     drag_and_drop: function (to, element) {
```
Comment (qsm-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: Also put the addons in tour_manager / tip.js / ... in another commit.

File: addons/web_tour/static/src/js/tour_manager.js [hunk 11]
```diff
@@ -133,13 +163,15 @@ return core.Class.extend({
...
+                self._to_next_step(name, 0);
+            }
 
-        if (!this.running_tour || this.running_tour === name) {
-            this.update(name);
-        }
+            if (self.active_tooltips[name] && (!self.running_tour || self.running_tour === name)) {
+                self.update(name);
+            }
+        });
```
Comment (qsm-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: Put this in a very separate commit, I am still not sure about it.

File: addons/web_tour/static/src/js/tour_manager.js [hunk 12]
```diff
@@ -331,7 +363,7 @@ return core.Class.extend({
             if (typeof tip.run === "function") {
                 tip.run.call(tip.widget, action_helper);
             } else if (tip.run !== undefined) {
-                var m = tip.run.match(/^(click|text|drag_and_drop) *(?:\(? *["']?(.+)["']? *\)?)?$/);
+                var m = tip.run.match(/^([a-zA-Z0-9_]+) *(?:\(? *(.+?) *\)?)?$/);
```
Comment (qsm-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: Why removing the ['"]? and what does (.+?) suppose to match ?

File: addons/website/static/src/js/website.editor.js [hunk 13]
```diff
@@ -216,7 +216,7 @@ widget.LinkDialog.include({
 /**
  * Display a welcome message on the homepage when it is empty and that the user is connected.
  */
-base.ready().then(function () {
+$.when(def, base.ready()).then(function () {
```
Comment (qsm-odoo) [inline addons/website/static/src/js/website.editor.js]: Isn't all ajax.loadXML supposed to be included in base.ready()? I think this file changes are useless.

File: addons/point_of_sale/static/src/js/tests.js (hunk 2 above)
Comment (Gorash) [inline addons/point_of_sale/static/src/js/tests.js]: It just check the dom, there is nothing to do

File: addons/web/static/test/x2many.js (hunk 3 above)
Comment (Gorash) [inline addons/web/static/test/x2many.js]: Ok, I did not look all pretty committed to make, that we do when the code is good

File: addons/web_tour/static/src/js/tour_manager.js (hunk 9 above)
Comment (Gorash) [inline addons/web_tour/static/src/js/tour_manager.js]: Yes, need for select, input (because trigger input but not change in automatic mode) and for content change (when edit content with editor)

File: addons/point_of_sale/static/src/js/tests.js (hunk 2 above)
Comment (qsm-odoo) [inline addons/point_of_sale/static/src/js/tests.js]: Yes but : 
Step 1 -> trigger found -> run click -> dom mutation -> Step 2 -> trigger found -> run empty (only a check) -> NO mutation -> ... the tour is blocked.
What am I missing? ^^

File: addons/web_tour/static/src/js/tour_manager.js (hunk 12 above)
Comment (Gorash) [inline addons/web_tour/static/src/js/tour_manager.js]: Don't need ['"] because we take all the text each time, and it's confusing if you want to match "text" and the result must contains ".
(.+?) is the smaller match pattern also the reset \)? does not work.

File: addons/website/static/src/js/website.editor.js (hunk 13 above)
Comment (Gorash) [inline addons/website/static/src/js/website.editor.js]: I'm looking for this but, they are a bug when we load this template on the website.

File: addons/web_tour/static/src/js/tour_manager.js (hunk 12 above)
Comment (qsm-odoo) [inline addons/web_tour/static/src/js/tour_manager.js]: Oh I did not know about the .+?, maybe this is worth to be in a separate commit as well as a fix.

File: addons/point_of_sale/static/src/js/tests.js (hunk 2 above)
Comment (Gorash) [inline addons/point_of_sale/static/src/js/tests.js]: not blocked... just consume nothing and continue

**Comment by Gorash** [comment]
I made the changes

File: openerp/addons/test_new_api/tests/test_ui.py [hunk 14]
```diff
@@ -1,10 +1,11 @@
...
 class TestUi(openerp.tests.HttpCase):
+    post_install = True
+    at_install = False
+
     def test_01_admin_widget_x2many(self):
         self.phantom_js("/web#action=test_new_api.action_discussions",
-                        "odoo.__DEBUG__.services['web.Tour'].run('widget_x2many', 'test')",
-                        "odoo.__DEBUG__.services['web.Tour'].tours.widget_x2many",
-                        login="admin")
+                        "odoo.__DEBUG__.services['web_tour.tour'].run('widget_x2many', 100)",
```
Comment (qsm-odoo) [inline openerp/addons/test_new_api/tests/test_ui.py]: I did not saw this ^^ 100 is needed ?

File: addons/web_tour/__openerp__.py [hunk 15]
```diff
@@ -13,7 +13,8 @@
     'depends': ['web'],
     'data': [
         'views/tour_templates.xml',
-        'views/tour_views.xml'
+        'views/tour_views.xml',
+        'security/ir.model.access.csv',
     ],
     'qweb': [
         "static/src/xml/*.xml",
```
Comment (qsm-odoo) [inline addons/web_tour/__openerp__.py]: In fact, why do we want tours for public user ? Was it not decided it is for very first admin only ?

File: addons/web_tour/static/src/js/tip.js [hunk 16]
```diff
@@ -259,7 +259,7 @@ var Tip = Widget.extend({
```
Comment (qsm-odoo) [inline addons/web_tour/static/src/js/tip.js]: Typo in message commit

File: addons/web_tour/__openerp__.py (hunk 15 above)
Comment (Gorash) [inline addons/web_tour/__openerp__.py]: website, ecommerce tours for e.g.
